models.cpp

```diff
@@ -554,37 +555,50 @@ std::valarray<double> Models::ThreeStepAlternative::right_hand_side(const std::v
   f[three_step_alt_parameters.n_variable - 1) = dn_{maxsize}
   */
   std::valarray<double> f(three_step_alt_parameters.n_variables);
-  // precursor -- loss from dissociative step, gain from dissociative step
-  f[0] = -three_step_alt_parameters.k_forward * x[0] * three_step_alt_parameters.solvent * three_step_alt_parameters.solvent + three_step_alt_parameters.k_backward * x[1]*x[2];
-  // dissasociated precursor -- opposite effect from the dissociative step as the precursor
-  f[1] = -f[0];
-  // precursor -- loss from nucleation
-  f[0] -= three_step_alt_parameters.k1 * x[0] * x[1] * x[1];
-  // dissasociated precursor -- loss from nucleation
+
+  // precursor
+    // loss from nucleation
+    // loss from particle growth
+    // loss from forward dissociative step
+    // gain from backward dissociative step
+  f[0] = -three_step_alt_parameters.k1 * x[0] * x[1] * x[1];
+  for (unsigned int i = 4; i < three_step_alt_parameters.n_variables; i++)
```

Comment:
This also looks like a change that you could isolate into a separate pull request and have it tested separately.